### PR TITLE
Support For PyTorch Neural Network Models

### PR DESCRIPTION
**Pull request by bpaul4**
## Fixes/Addresses:

(replace this with the issue # fixed or addressed, if no issue exists then a brief statement of what this PR does)

## Changes proposed in this PR:
- Demonstrates how users can train/save PyTorch neural networks with custom layer objects and produce .pt files compatible with the ML AI Plugin in FOQUS
- Support for Pytorch models in FOQUS, including updates to the class structure, search method, example files, documentation, tests, and optional dependencies list

### Legal Acknowledgement

By contributing to this software project, I agree to the following terms and conditions for my contribution:

1. I agree my contributions are submitted under the copyright and license terms described in the LICENSE.md file at the top level of this directory.
2. I represent I am authorized to make the contributions and grant the license. If my employer has rights to intellectual property that includes these contributions, I represent that I have received permission to make contributions and grant the required license on behalf of that employer.

### CODE REVIEW

**Comment by ksbeattie** [comment]
@anujad95 & @Daison2102 can you review this in time for Dec release?  We're hoping to cut the release candidate on Nov29th with this PR included as part of a milestone due in Dec.

**Review by Daison2102 — DISMISSED** [review]
LGTM. 
Pytorch related test passed and example run smoothly

File: examples/other_files/ML_AI_Plugin/mea_column_model_training_customnormform_pytorch.py [hunk 1]
```diff
@@ -0,0 +1,163 @@
+import numpy as np
+import pandas as pd
+import random as rn
+import torch
+import torch.nn as nn
+import math
```
[the remark this answers is not in the document]
Comment (bpaul4) [reply examples/other_files/ML_AI_Plugin/mea_column_model_training_customnormform_pytorch.py]: Done, and for the other scripts as well. Thanks for catching this.

**Review by Daison2102 — APPROVED** [review]
LGTM

**Pull request merged**